### PR TITLE
174917481 fix theme css loading development

### PR DESCRIPTION
**Pull request by knowuh**
@scytacki 

This has got the major functionality working. I have verified that pre-compile and development works as expected with the NGSS theme. 

There was a lot of cleanup, and there is potentially a much larger blast radius than I would like.

There is more cleanup todo to hunt down all the uses of APP_CONFIG[:theme] and replace them with ENV['THEME'].  I could spend a few more hours on that probably.

### CODE REVIEW

**Comment by knowuh** [comment]
Also want to see how builds go.

File: rails/app/themes/xproject/assets/stylesheets/xproject/all.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*
+ *= require_self
```
Comment (scytacki) [inline rails/app/themes/xproject/assets/stylesheets/xproject/all.scss:2]: This should be removed too right?

**Review by scytacki — COMMENTED** [review]
looks good so far

**Comment by knowuh** [comment]
The build fails on Travis with 

```
Error: error in C function image-url: undefined method `[]' for nil:NilClass
2523        on line 12:27 of stdin, in function `image-url`
2524        from line 12:27 of stdin
2525>>   background: transparent image-url("/assets/banners/ngss-assessment.png") 0
2526   --------------------------^
```

This while sprockets is building the NGSS stylesheets in the rake task:  
`RAILS_GROUPS=assets ./bin/rake assets:precompile`

The same build process works fine locally, and produces a compiled NGSS stylesheet
(from inside the rails directory on the docker container):

```
  rm -rf /rigse/public/assets/
  export RAILS_GROUPS=assets
  bundle exec rake assets:precompile
```

**Comment by scytacki** [comment]
I found this: https://github.com/sass/sassc-rails/issues/121
Which refers to this:
https://github.com/rails/sprockets/issues/581#issuecomment-486984663

So maybe we can try disabling concurrency.

**Comment by knowuh** [comment]
I had read that too, but our version of sprockets its 3.7.2, and the option for `export_concurrent` might be only for versions 4.x+ — So I am thinking about trying an update of sprockets and the `sassc` gems.

**Comment by knowuh** [comment]
@scytacki that concurrency thing seems to have been a false alarm.  Instead, the NGSS stylesheet filename was incorrect (`.css`), and it wasn't being processed by SASS as it should have been.  After fixing that in 538c880 tests passed.

My question now is whether to undo the SASS Gem update and `disable_concurrency` configuration change in 68bf06c.

I guess my feeling is not leave those changes in, but I value your thoughts on that.

I would like to make a docker image now to try things out in the QA environment.

**Comment by scytacki** [comment]
Yeah I'd take out the concurrency thing if it isn't needed.

File: rails/app/themes/xproject/assets/stylesheets/xproject/all.scss (hunk 1 above)
Comment (scytacki) [reply rails/app/themes/xproject/assets/stylesheets/xproject/all.scss:2]: @knowuh how about this file, should it be removed?

**Review by scytacki — APPROVED** [review]
Looks good.

**Pull request merged**